rho/profileclearcommand.py

```diff
@@ -61,75 +82,43 @@ def _validate_options(self):
-                    del_host_auth_mapping = '(DELETED PROFILE)' + profile \
-                        + '_host_auth_mapping'
-                    del_host_auth_mapping_path = \
-                        get_config_path(del_host_auth_mapping)
-                    os.rename(profile_mapping,
-                              del_host_auth_mapping_path)
-            else:
-                print(_("All network profiles removed"))
+            vault = get_vault(self.options.vaultfile)
+            profile = self.options.name
+            profiles_list = vault.load_as_json(utilities.PROFILES_PATH)
+            profile_found = False
+
+            for index, curr_profile in enumerate(profiles_list):
+                if curr_profile.get('name') == profile:
+                    del profiles_list[index]
+                    print(_('Profile "%s" was removed' % profile))
+                    profile_found = True
+                    break
+
+            if not profile_found:
+                print(_("No such profile: '%s'") % profile)
+                sys.exit(1)
+
+            vault.dump_as_json_to_file(profiles_list, utilities.PROFILES_PATH)
+
+            # removes inventory associated with the profile
+            profile_hosts_path = get_config_path(profile + PROFILE_HOSTS_SUFIX)
+            if os.path.isfile(profile_hosts_path):
+                os.remove(profile_hosts_path)
+            _backup_host_auth_mapping(profile)
 
         # removes all inventories ever.
         elif self.options.all:
-            if not os.path.isfile(utilities.PROFILES_PATH):
-                print(_("All network profiles removed"))
-            else:
-                os.remove(utilities.PROFILES_PATH)
-                wildcard_hosts = '*_hosts'
-                wildcard_hosts_path = get_config_path(wildcard_hosts)
-                for file_list in glob.glob(wildcard_hosts_path):
-                    os.remove(file_list)
-                    profile = file_list.strip('_hosts')
-                    host_auth_mapping = profile + '_host_auth_mapping'
-                    profile_mapping = get_config_path(host_auth_mapping)
-                    if os.path.isfile(profile_mapping):
-                        del_host_auth_mapping = '(DELETED PROFILE)' \
-                            + profile + '_host_auth_mapping'
-                        del_host_auth_mapping_path = \
-                            get_config_path(del_host_auth_mapping)
-                        os.rename(profile_mapping,
-                                  del_host_auth_mapping)
-
-                print(_("All network profiles removed"))
+            os.remove(utilities.PROFILES_PATH)
+            wildcard_hosts_path = get_config_path('*' + PROFILE_HOSTS_SUFIX)
+            for file_list in glob.glob(wildcard_hosts_path):
+                os.remove(file_list)
+                file_list = os.path.basename(file_list)
+                profile = file_list[:file_list.rfind(PROFILE_HOSTS_SUFIX)]
```

Comment:
This remove the suffix as expected:

```python
>>> path = 'test_hosts.yml'
>>> path[:path.rfind('_hosts.yml')]
'test'
```

Another option is to use the `re` module. That said, let me know which one you think is easier to understand.